lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Trento.DeregistrationProcessManager do
+  @moduledoc """
+    DeregistrationProcessManager is a Commanded ProcessManager, it's the responsible
+    for the deregistration procedure for both hosts and clusters.
```

Comment:
I would maybe remove this `hosts and clusters` or in any case include the sapsystems.
Maybe simply put `deregistration procedure for the aggregate, such as hosts, clusters, sap systems, etc